src/vellum/workflows/nodes/displayable/tool_calling_node/utils.py

```diff
@@ -287,7 +287,12 @@ def create_port_condition(fn_name):
 
     # Add a chat history block to blocks only if one doesn't already exist
     has_chat_history_block = any(
-        block.block_type == "VARIABLE" and block.input_variable == CHAT_HISTORY_VARIABLE for block in blocks
+        (
+            (block.get("block_type") if isinstance(block, dict) else getattr(block, "block_type", None)) == "VARIABLE"  # type: ignore  # noqa: E501
+            and (block.get("input_variable") if isinstance(block, dict) else getattr(block, "input_variable", None))  # type: ignore  # noqa: E501
```

Comment:
[nit] should be able to just return `block.block_type` instead of `getattr`. Why the type ignore?